madara/crates/client/settlement_client/src/starknet/mod.rs

```
@@ -65,20 +67,25 @@ impl StarknetClient {
             Felt::from_hex(&config.core_contract_address).map_err(|e| -> SettlementClientError {
                 StarknetClientError::Conversion(format!("Invalid core contract address: {e}")).into()
             })?;
-        // Check if l2 contract exists
-        provider.get_class_at(BlockId::Tag(BlockTag::Latest), core_contract_address).await.map_err(
-            |e| -> SettlementClientError {
-                StarknetClientError::NetworkConnection { message: format!("Failed to connect to L2 contract: {}", e) }
-                    .into()
-            },
-        )?;
+
+        // Note: We no longer check if the contract exists here to avoid blocking startup
```

same as earlier, I don't think we should remove this check at the begining

---

Fixed! Added contract verification at startup using get_class_hash_at(). If the contract doesn't exist, it fails fast with a clear error message instead of retrying indefinitely.